distributions/nrdot-collector-host/assert-invariants-agent-control.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+set -e
+yq --version | grep -E -q '4\.[0-9]+\.[0-9]+' ||
+  { echo $? && echo "yq version 4.x.x is expected, but was '$(yq --version)'"; exit 1; }
+
+script_name=$(basename "$0")
+script_dir=$(realpath "$(dirname "${BASH_SOURCE[0]}")")
+host_distro_dir="${script_dir}"
+
+# assert config file location
+test -f "${host_distro_dir}/config.yaml" ||
+  { echo "expect config at config.yaml"; exit 1; }
+
+# assert unchanged env vars
+env_vars=(
+  'NEW_RELIC_MEMORY_LIMIT_MIB'
+  'OTEL_EXPORTER_OTLP_ENDPOINT'
+  'NEW_RELIC_LICENSE_KEY'
+)
+for env_var in "${env_vars[@]}"; do
+  echo "Checking for env var ${env_var}"
+  echo "\${env:${env_var}[^}]+}"
+  grep -E '\${env:[^}]+}' "${host_distro_dir}/config.yaml" |
+  grep "${env_var}" ||
+    { echo "expected env var '${env_var}' in config.yaml"; exit 1; }
+done
+# TODO: assert additional invariants within config
+
+
+# assert binary name
+goreleaser_yamls=('.goreleaser.yaml' '.goreleaser-nightly.yaml')
+for goreleaser_yaml in "${goreleaser_yamls[@]}"; do
+  echo "Checking ${goreleaser_yaml}"
+  yq -e '.builds[].binary == "nrdot-collector-host"' "${host_distro_dir}/${goreleaser_yaml}" ||
+    { echo "expected binary name 'nrdot-collector-host' in ${goreleaser_yaml}"; exit 1; }
```

Comment:
Awesome that you fail the pipeline, we are gonna ensure we don't break the contract.